src/NLog/LogFactory.cs

```diff
@@ -1260,11 +1260,11 @@ private void ConfigFileChanged(object sender, EventArgs args)
         /// <summary>
         /// Logger cache key.
         /// </summary>
-        internal class LoggerCacheKey : IEquatable<LoggerCacheKey>
+        internal struct LoggerCacheKey : IEquatable<LoggerCacheKey>
```

Comment:
Looks very much in line:

- It logically represents a single value (Its a key)
- It has an instance size under 16 bytes (check)
- It is immutable (check)
- It will not have to be boxed frequently (check)